src/window/User Guide.md

```diff
@@ -0,0 +1,132 @@
+% User Guide to Using Piston Windows
+
+# Guide to using Piston Windows
+
+## Getting Help
+
+*Notice! Piston is being actively developed. Please let us know if anything in this tutorial is outdated.*
+
+See the main [Piston Guide](../../GUIDE.md) for general help with Piston.
+
+## Overview
+
+The piston engine is designed to be an abstraction. It provides the same functionality through several different backends, each of which have their own quirks and extra features.
+
+Because of this abstraction, some of the details of the piston API may seem unnecessary or useless. The purpose of this guide is to give concrete guidelines on the creation and usage of piston windows.
```

Comment:
"unnecessary or useless"? Why should an abstraction seem that way?

I know that the word "abstraction" has lots of hidden connotations, some related to "complex", "air castle" etc. However, the clearest meaning of the word "abstraction" is simply this:

> abstraction = reuse

For example, a function is an abstraction because it lets you reuse computer instructions.

The Piston core is an abstraction for reusing code. It the minimum amount of code that lets us reuse as much as possible code between projects.